### ISSUE TITLE
Type inference of Collections Type

### ISSUE THREAD

**Issue by zsh1995** [microsoft/pyright#473]
**Describe the bug**
When I user a Dict contains different subclasses which hava same superclass，pyright inference Dict's value as Union Type:  
```python
class A:
  pass
class B(A):
  pass
class C(A):
  pass

_T_A = TypeVar('_T_A', bound=A)

def testFunc(value: Dict[str, _T_A]):
  pass

testFunc({
  "b": B(),
  "c": C()
}) # error
```
errors like this
```
Argument of type 'Dict[str, Union[B, C]]' cannot be assigned to parameter 'value' of type 'Dict[str, _T_A]'
  Type 'Union[B, C]' is not compatible with bound type 'A' for TypeVar '_T_A'
    Type 'B' cannot be assigned to type 'A'
    Type 'C' cannot be assigned to type 'A'
      'B' is incompatible with 'A'
      'C' is incompatible with 'A'pyright
```
If I assign in this way, it works:
```python
param: Dict[str, _T_A]{
  "b": B(),
  "c": C()
}
testFunc(param)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of Pyright.

**Comment by erictraut** [comment]
This is implemented in 1.1.18, which I just published.

**Comment by zsh1995** [comment]
Thanks a lot @erictraut
